newsroom/agenda/views.py

```diff
@@ -287,3 +287,145 @@ def related_wire_items(wire_id):
         ),
         200,
     )
+
+
+@blueprint.route("/agenda/search_locations")
+@login_required
+def search_locations():
+    query = request.args.get("q") or "*"
+
+    if not query.startswith("*") and not query.endswith("*"):
+        query = f"*{query}*"
+
+    def gen_agg_filter(field: str):
+        return {
+            "bool": {
+                "must": [
+                    {
+                        "query_string": {
```

Comment:
might be better to use `match` query here or `match_bool_prefix`

Reply:
Using `match` or `match_bool_prefix` requires to normalise the terms, as to provide case insensitive search. These searches also do not allow to partial match (only `startswith` match), where as `query_string` provides this and much more (although most functionality won't be used by the general public users).

To improve performance, I have removed the `query_string` filters if there is no search text provided.